### PR TITLE
fix: handle invalid chart configuration

### PR DESCRIPTION
**Pull request by ZeRego**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #1927 

### Description:

- avoid visualization provider from recalculating all the configuration when results are refreshed ( since `resultsData` is undefined while is loading )
- avoid setting multiple times the chart configuration by using `else if` 
- allow the chart configuration to set fallback axes when results data changes

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

File: packages/frontend/src/hooks/useCartesianChartConfig.ts [hunk 1]
```diff
@@ -190,33 +190,37 @@ const useCartesianChartConfig = (
...
         const setAxes = (
             prev: Partial<Partial<CompleteCartesianChartLayout>> | undefined,
-            xField: string,
-            yField: string[],
+            xField: string | undefined,
+            yFields: string[],
         ) => {
-            const validYFields = prev?.yField
+            const isCurrentXFieldValid: boolean =
+                !!prev?.xField && availableFields.includes(prev.xField);
```
Comment (rephus) [inline packages/frontend/src/hooks/useCartesianChartConfig.ts:197]: don't we need to check here if `xField !== undefined` ?

**Review by rephus — COMMENTED** [review]
Code looks good to me. But I can still get some invalid charts when selecting/deselecting fields. In  a few cases both axes were using the same dimension..
Peek 2022-05-20 09-09

File: packages/frontend/src/hooks/useCartesianChartConfig.ts (hunk 1 above)
Comment (ZeRego) [reply packages/frontend/src/hooks/useCartesianChartConfig.ts:197]: I don't think so. We should allow the xField to be changed to undefined if there aren't any available fields. 

E.g: User had xField and yFields defined. Then unselected the field used in xField. We need to set the xField to undefined.

**Comment by ZeRego** [comment]
> Code looks good to me. But I can still get some invalid charts when selecting/deselecting fields. In a few cases both axes were using the same dimension.

True! let me see if I can change the logic to prevent this. :)

**Comment by ZeRego** [comment]
@rephus Had a chat with @TuringLovesDeathMetal about how we should update the chart configuration if it is partially invalid. Updated the notion doc with the logic and priorities: https://www.notion.so/lightdash/Default-chart-configurations-5d3001af990d4b6fa990dba4564540f6

**Pull request merged**